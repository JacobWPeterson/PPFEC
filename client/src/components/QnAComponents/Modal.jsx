import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

const StyledModal = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 2000;
  width: 800px;
  height: 100%;
  margin: 40px auto;
  transform: translate(-50%, -50%);
`;

// `
//   background-color: #424242;
//   opacity: 0.6;
//   position: absolute;
//   top: 0;
//   left: 0;
//   right: 0;
//   bottom: 0;
//   display: flex;
//   align-items: center;
//   justify-content: center;
//   z-index: 200;
//   width: 100%;
//   height: 100%;
//   margin: 40px auto;
// `;
class Modal extends React.Component {
  render() {
    return ReactDOM.createPortal(
      <div>
        {/* <DarkLayer /> */}
        <StyledModal>
          <div>
            {this.props.content}
          </div>
        </StyledModal>
      </div>,
      document.getElementById('modal-root') || document.createElement('div'),
    );
  }
}

export default Modal;
