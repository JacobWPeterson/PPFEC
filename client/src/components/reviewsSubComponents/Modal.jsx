import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

const StyledModal = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 2000;
  width: 800px;
  margin: 40px auto;
`;

const DarkLayer = styled.div`
  background-color: black;
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 200;
  width: 100%;
  margin: 40px auto;
`;
class Modal extends React.Component {
  render() {
    return ReactDOM.createPortal(
      <div>
        <DarkLayer />
        <StyledModal>
          <div>
            {this.props.content}
          </div>
        </StyledModal>
      </div>,
      document.getElementById('modal-root'),
    );
  }
}

export default Modal;
