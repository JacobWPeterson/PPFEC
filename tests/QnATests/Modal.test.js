import React from 'react';
import { mount } from 'enzyme';
import Modal from '../../client/src/components/QnAComponents/Modal.jsx';

describe('This is for the Modal', () => {
  const modalComp = mount(<Modal />);

  it('Should render More Modal', () => {
    expect(modalComp).toMatchSnapshot();
  });
});
