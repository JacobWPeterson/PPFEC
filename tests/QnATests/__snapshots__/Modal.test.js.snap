// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is for the Modal Should render More Modal 1`] = `
<Modal>
  <Portal
    containerInfo={
      <div>
        <div>
          <div
            class="Modal__StyledModal-sc-1od4ot6-0 hYuCAL"
          >
            <div />
          </div>
        </div>
      </div>
    }
  >
    <div>
      <Modal__StyledModal>
        <div
          className="Modal__StyledModal-sc-1od4ot6-0 hYuCAL"
        >
          <div />
        </div>
      </Modal__StyledModal>
    </div>
  </Portal>
</Modal>
`;
